# Variabilidade

## Dispersão / Variação

A variação pode ser calculada subtraindo o valor maximo da variavel do eixo X com o valor minimo do eixo X

Ex: 
- Maior salario 78,600
- Menor salario 21,180
- A variacao é 57,420

## Quartil

E quando pegamos a variação e dividimos ela por 4

ex.
1
2 - q1
3 
4 - q2
5
6 - q3
7
8 - q4

## IQR Interquartil Range

E a variação entre o range dos quartis, usando os dados acima, o primeiro e o ultimo quartil são cortados ficando somente o 2 e o 3.

o IQR seria 3, pois o segundo quartil começa em 3 e o terceiro quartil termina em 6, então 6 - 3 = 3

## Outlier

Para definir se um número é ou não um outlier tem uma formula padrão que podemos usar.

- outlier < q1 - 1,5 * IQR
- outlier > q3 + 1,5 * IQR

Calculando Outliers

Q1 = 49191
Q3 = 54135
IQR = 4944

In [4]:
49191 - (1.5 * 4944)

41775.0

In [6]:
54135 + (1.5 * 4944)

61551.0

Com isso temos que números abaixo de 41775 e números acima de 61551 serão considerados outliers pelas amostra.


## Desvio da média

Para calcular o desvio da média usamos (xi - média)

Depois para calcularmos o desvio médio precisamos retirar os números negativos para que eles não se anulem com os valores positivos em relação a média

A duas maneiras de retirarmos esses números negativos

- Ignorando o sinal negativo
- Elevando ao quadrado os números


## Variancia

Como podemos definir o conceito de variância:

- Média dos quadrados dos desvios
- Soma dos quadrados dos desvios dividida por n

## Desvio Padrão

De que maneira poderíamos definir o desvio padrão?

- Raiz quadrada da média do quadrado do desvio
- Raiz quadrada da soma do quadrado do desvio, dividida por n

Quando vou calcular o desvio padrão de uma população uso a media da soma dos quadrados dos desvios
Quando vou calcular o desvio padrão de uma amostra, uso a correção de bessel na formula.


Passos para calcular o desvio padrão

1. Média dos valores da amostra
2. Desvio da média (xi - média)
3. Quadrado do desvio da média
4. Média do quadrado do desvio
5. Raiz quadrada da média do quadrado do desvio

Em uma distribuição normal a média, mediana e moda são iguais

Quando temos uma distribuição normal, 68% dos dados ficam entre a média e 1 desvio padrão, e 95% dos dados ficam entre a média e 2 desvis padrões.



In [15]:
import numpy as np
# np.array([33.219,36.254,38.801,46.335,46.840,47.596,55.130,56.863,78.070,88.830]).mean()

# np.array([-19574.80,-16539.80,-13992.80,-6458.80,-5953.80,-5197.80,2336.20,4069.20,25276.20,36036.20]).mean()

np.array([19574.80,16539.80,13992.80,6458.80,5953.80,5197.80,2336.20,4069.20,25276.20,36036.20]).mean()

13543.559999999998

In [6]:
78600 - 21180

57420

In [2]:
116020 - 7350

108670

In [1]:
10000000 - 21180

9978820

In [3]:
54135 - 49191

4944

In [17]:
from math import sqrt
sqrt(291622740)

17076.965187058267